### DISCUSSION TITLE
Error Initializing `CodeSherpa.js` Despite Exclusion from Included Tools Config

### DISCUSSION THREAD

**Discussion by Guru-25** [danny-avila/LibreChat#4083] [Troubleshooting] [answered]
### What happened?

After excluding `CodeSherpa.js` from the includedTools configuration, the system is still attempting to initialize the tool. This results in an error message: `Error initializing CodeSherpa.js tool; if it requires authentication, is the override field configured? Missing CODESHERPA_SERVER_URL environment variable.`

I expected the `CodeSherpa.js` tool to be disabled completely after exclusion from the configuration, so no initialization attempt or error related to missing environment variables should have occurred.

### Steps to Reproduce

1. In `librechat.yaml`, add the following configuration:
   ```yaml
   includedTools: ["google"]
   ```
2. Start or reload the application.
3. Observe the following error message:
   ```
   Error initializing CodeSherpa.js tool; if it requires authentication, is the override field configured? Missing CODESHERPA_SERVER_URL environment variable.
   ```

4. Now, edit `librechat.yaml` and remove the `includedTools` field:
   ```yaml
   # includedTools: ["google"]
   ```
5. Start or reload the application again.
6. Notice that the error is no longer shown.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by RAPHCVR** [comment]
Same error here when not including this tool in the includedtools only.

**Comment by danny-avila** [comment, marked as answer]
I've unintentionally excluded non-structured tools from the list of tools that render in the frontend, will be fixing that soon.
